test/e2e/e2e.go

```diff
@@ -188,14 +189,48 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 
 })
 
+var lock sync.Mutex
+var cleanupActions = map[*int]func(){}
+
+func AddCleanupAction(fn func()) *int {
+	p := new(int)
```

Comment:
Is there really no better way to get a set object (which is what, as far as I can tell, you're doing).